numpyro/infer/autoguide.py

```diff
@@ -864,6 +865,373 @@ def _single_sample(_rng_key):
             return _single_sample(rng_key)
 
 
+def _subsample_model(model, *args, **kwargs):
+    data = kwargs.pop("_subsample_idx", {})
+    with handlers.substitute(data=data):
+        return model(*args, **kwargs)
+
+
+class AutoSemiDAIS(AutoGuide):
+    """
+    This implementation of :class:`AutoSemiDAIS` [1] combines a parametric
+    variational distribution over global latent variables with Differentiable
+    Annealed Importance Sampling (DAIS) [2, 3] to infer local latent variables.
+    Unlike :class:`AutoDAIS` this guide can be used in conjunction with data subsampling.
+    Note that the resulting ELBO can be understood as a particular realization of a
+    `locally enhanced bound' as described in reference [4].
+
+    **References:**
+
+    1. *Surrogate Likelihoods for Variational Annealed Importance Sampling*,
+       Martin Jankowiak, Du Phan
+    2. *MCMC Variational Inference via Uncorrected Hamiltonian Annealing*,
+       Tomas Geffner, Justin Domke
+    3. *Differentiable Annealed Importance Sampling and the Perils of Gradient Noise*,
+       Guodong Zhang, Kyle Hsu, Jianing Li, Chelsea Finn, Roger Grosse
+    4. *Variational Inference with Locally Enhanced Bounds for Hierarchical Models*,
+       Tomas Geffner, Justin Domke
+
+    Usage::
+
+        def global_model():
+            return numpyro.sample("theta", dist.Normal(0, 1))
+
+        def local_model(theta):
+            with numpyro.plate("data", 8, subsample_size=2):
+                tau = numpyro.sample("tau", dist.Gamma(5.0, 5.0))
+                numpyro.sample("obs", dist.Normal(0.0, tau), obs=jnp.ones(2))
+
+        model = lambda: local_model(global_model())
+        base_guide = AutoNormal(global_model)
+        guide = AutoSemiDAIS(model, local_model, base_guide, K=4)
+        svi = SVI(model, guide, ...)
+
+    :param callable model: A NumPyro model.
+    :param callable local_model: The portion of `model` that includes the local latent variables only.
```

Comment:
are these the right/complete requirements for local_model and base_guide?

Reply:
I think so. Which information else that you have in mind?